### PR TITLE
feat: Define ArgentX theme and make it the single-source of truth

### PR DESCRIPTION
**Pull request by dhruvkelawala**

### CODE REVIEW

File: packages/extension/src/ui/theme/index.tsx [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  css,
+} from "styled-components"
+import { normalize } from "styled-normalize"
+
+import { Colors } from "./styled"
+
+const white = "#FFFFFF"
+const black = "#000000"
+
+export const colors = (): Colors => {
```
Comment (simonheys) [inline packages/extension/src/ui/theme/index.tsx:16]: What's the reason for this to be a function? Expect to pass some parameters to it in future?

File: packages/extension/src/ui/theme/index.tsx [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+const MEDIA_WIDTHS = {
+  sm: 600,
+  md: 900,
+  lg: 1200,
+  xl: 1536,
+}
+
+const mediaMaxWidthTemplates: {
```
Comment (simonheys) [inline packages/extension/src/ui/theme/index.tsx:51]: This is clever, maybe too clever for me so I also find it hard to reason about. Can you add comments at least that show the expected shape of the templates and how they would be used?

File: packages/extension/src/ui/theme/styled.d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { FlattenSimpleInterpolation } from "styled-components"
+
+export type Color = string
+export interface Colors {
```
Comment (simonheys) [inline packages/extension/src/ui/theme/styled.d.ts:4]: Why is this replicating the shape of colors function? Could be using e.g. typeof colors if colors was not a function?

File: packages/extension/src/ui/theme/styled.d.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  sm: ThemedCssFunction<DefaultTheme>
+  md: ThemedCssFunction<DefaultTheme>
+  lg: ThemedCssFunction<DefaultTheme>
+  xl: ThemedCssFunction<DefaultTheme>
+}
+
+declare module "styled-components" {
+  export interface DefaultTheme extends Colors {
+    // css snippets
+    flexColumnNoWrap: FlattenSimpleInterpolation
```
Comment (simonheys) [inline packages/extension/src/ui/theme/styled.d.ts:47]: Would be great to see more comments or stories around how these are expected to be used

File: packages/extension/src/ui/theme/styled.d.ts (hunk 3 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/theme/styled.d.ts:4]: It's the other way around. This defines the structure/shape of `colors`. And I agree, Colors doesn't have to be function as of now. Might only be useful when we introduce lightMode/darkMode.

File: packages/extension/src/ui/theme/styled.d.ts (hunk 4 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/theme/styled.d.ts:47]: Added better docs

File: packages/extension/src/ui/theme/index.tsx (hunk 2 above)
Comment (dhruvkelawala) [reply packages/extension/src/ui/theme/index.tsx:62]: Added docs

File: packages/extension/src/ui/theme/styled.d.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { FlattenSimpleInterpolation } from "styled-components"
+
+export type Color = string
```
[the remark this answers is not in the document]
Comment (dhruvkelawala) [reply packages/extension/src/ui/theme/styled.d.ts:3]: I don't agree with you on this one. `colors` object should be derived from `Colors` type and not the other way around. That's what typings are for. I don't understand what's the point of doing it the other way around. We always make an object from a type and not type from an object.

**Review by simonheys — APPROVED** [review]

**Comment by dhruvkelawala** [comment]
@simonheys Using `typeof` colors

**Pull request merged**